wandb/sync/sync.py

```diff
@@ -117,7 +119,9 @@ def _find_tfevent_files(self, sync_item):
                     if tb_dir not in tb_logdirs:
                         tb_logdirs.append(tb_dir)
                 if len(tb_logdirs) > 0:
-                    tb_root = os.path.commonprefix(tb_logdirs)
+                    tb_root = to_forward_slash_path(
```

Comment:
Is this ok on windows?  This means we can't use tb_root as an actual file path.

Reply:
Oh I can swap this out to a windows friendly version. The issue with just commonprefix was that if we had 2 directories, one "test" and the other "train" we got sections "est" and "rain" in the UI, because the "t" was common.

Reply:
I'm confused, how does this code prevent that?

Reply:
Oh darn, yeah sorry thats the point of os.path.dirname. The purpose of this was to make it work on windows. It's meant to mirror how we calculate the rootdir when we calculate the namespace as in tb_watcher. I'll reinvestigate that code path to make sure its right.

Reply:
Still concerned about this directory not being readable in Windows.

Reply:
I'm confused here, this `to_forward_slash_path` is defined to convert things to windows:
```
def to_forward_slash_path(path):
    if platform.system() == "Windows":
        path = path.replace("\\", "/")
    return path
```
won't this function handle it?

Reply:
Windows paths look like `C:\\foo\bar\baz`.  To forward slash converts backslashes to forward slashes so this would become `C://foo/bar/baz`.  This is how the paths are stored in cloud storage.  If you try to read a path generated by `to_forward_slash_path` on windows it likely won't work.

Reply:
Ah, got it. Removed